nexus/src/external_api/console_api.rs

```diff
@@ -260,7 +270,7 @@ pub async fn login(
                                     .to_str()
                                     .map_err(|e| {
                                         HttpError::for_internal_error(format!(
-                                            "{}",
+                                            "referer header to_str failed! {}",
```

Comment:
super nitty but I imagine this should be a 400 for providing an unsupported header?

Reply:
Yeah, that makes sense. I originally thought that the user wouldn't control the referer header, but of course they do, because they navigate to a URL in the first place. Changed in d66038f4